### PR TITLE
Contributions: "give to the Guardian" header with coin icon

### PR DESCRIPTION
**Pull request by rtyley**
## What does this change?

Testing a "give to the Guardian" header, which replaces 'become a member' in the nav bar. Links to https://contribute.theguardian.com/
## Does this affect other platforms - Amp, Apps, etc?

Nope.
## Screenshots

With the coin
image

without (ie normal view):
image
## Request for comment

cc @markjamesbutler @desbo

### CODE REVIEW

File: common/app/conf/switches/ABTestSwitches.scala [hunk 1]
```diff
@@ -63,4 +63,14 @@ trait ABTestSwitches {
     sellByDate = new LocalDate(2016, 8, 31),
     exposeClientSide = true
   )
+
+  val ABGiraffeHeader20160802 = Switch(
+    SwitchGroup.ABTests,
+    "ab-giraffe-header-20160802",
+    "Test effectiveness of header for driving contributions.",
+    owners = Seq(Owner.withGithub("markjamesbutler")),
+    safeState = On,
```
Comment (markjamesbutler) [inline common/app/conf/switches/ABTestSwitches.scala]: Let's change this to Off just in case.

File: static/src/inline-svgs/contributions/coin.svg [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 16.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Comment (gtrufitt) [inline static/src/inline-svgs/contributions/coin.svg]: You should run this through https://jakearchibald.github.io/svgomg/

Comment (rtyley) [inline static/src/inline-svgs/contributions/coin.svg]: Will do - I'm a bit surprised though, https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md#image-preparation says that inline SVGs are run through SVGO anyway, but they don't seem to be?

**Comment by zeftilldeath** [comment]
Maybe the coin could be the full circle size with no outline around it. Would be larger, then and I think it looks a bit strange being a circle within a circle. Our roundel does the same.

File: static/src/inline-svgs/contributions/coin.svg (hunk 2 above)
Comment (gtrufitt) [inline static/src/inline-svgs/contributions/coin.svg]: Hmh. One to add to our list...

File: static/src/javascripts/projects/common/modules/experiments/ab.js [hunk 3]
```diff
@@ -27,7 +28,8 @@ define([
     Giraffe,
```
Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/experiments/ab.js]: is `Giraffe` and `giraffeHeader` the clearest names for this? I know that your team name is that, but your team name may change and this code may still be here

File: static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        this.showForSensitive = false;
+        this.audience = 0.10;
+        this.audienceOffset = 0;
+        this.successMeasure = 'Determine if header is a good mechanism for driving contributions.';
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = '';
+        this.idealOutcome = '';
+        this.canRun = function () {
+            var pageObj = window.guardian.config.page;
+            return !(pageObj.isSensitive || pageObj.isLiveBlog || pageObj.isAdvertisementFeature) && pageObj.edition === 'UK';
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js]: Do you need to check `isSensitive` if you've set `showForSensitive` to` `false`?

File: static/src/inline-svgs/contributions/coin.svg [hunk 5]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="18" height="18" viewBox="5 5 18 18"><defs><circle id="a" cx="14" cy="14" r="9"/></defs><use xlink:href="#a" overflow="visible" fill="#FB0"/><clipPath id="b"><use xlink:href="#a" overflow="visible"/></clipPath><circle clip-path="url(#b)" fill="none" stroke="#FF9B0B" stroke-miterlimit="10" stroke-dasharray="2,2" cx="14" cy="14" r="7.615"/><g clip-path="url(#b)"><defs><path id="c" d="M11.64 14.604h.655v.593c0 .86-.296 1.237-.818 1.515v.89h4.902l.142-1.217h-3.478v-.03c.45-.33.726-.676.777-1.484l.01-.265h1.75v-.942h-1.687l.05-.665c.072-1.104.623-1.35 1.33-1.35.46 0 .798.07 1.145.194v-.962c-.39-.296-.89-.48-1.75-.48-1.36 0-2.373.777-2.373 2.404v.86h-.654v.94z"/></defs><clipPath id="d"><use xlink:href="#c" overflow="visible"/></clipPath><g clip-path="url(#d)"><defs><path id="e" d="M-55.676-2.755h986.54v542.77h-986.54z"/></defs><clipPath id="f"><use xlink:href="#e" overflow="visible"/></clipPath><path clip-path="url(#f)" fill="#C05303" d="M8.016 6.937h11.967v14.126H8.016z"/></g></g></svg>
```
Comment (NataliaLKB) [inline static/src/inline-svgs/contributions/coin.svg]: Please follow https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md and get rid off all the `xmlns` things

File: static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            });
+        };
+
+        this.variants = [
+            {
+                id: 'control',
+                test: function () {
+                    writer('give to the Guardian', 'https://contribute.theguardian.com/uk?INTCMP=co_uk_head_give_coin');
+                },
+                success: function (complete) {
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js]: `success: completer` would be the same?

File: static/src/javascripts/projects/common/modules/experiments/ab.js (hunk 3 above)
Comment (rtyley) [inline static/src/javascripts/projects/common/modules/experiments/ab.js]: Yeah, I'd be happy to change 'giraffe'-> 'contributions', though this PR is only to add the new header, so shall I just fix the `giraffeHeader` for the time being?

File: static/src/javascripts/projects/common/views/giraffe-header.html [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<div class="brand-bar__item brand-bar__item--give-to-the-guardian
+                                    brand-bar__item--has-control
+                                    popup-container has-popup
+                                    hide-until-mobile-landscape">
+    <a href="<%= linkHref %>"
+       id="giraffe__header-contribute-link"
+       data-link-name="give to the Guardian"
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/views/giraffe-header.html]: Might be worth making this a bit more descriptive eg: `header : give to the Guardian` to differentiate it from other place

File: static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js (hunk 6 above)
Comment (desbo) [inline static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js]: since there's only one use I'd just define the function at the property

**Comment by piuccio** [comment]
looks kinda like the lidl logo, yellow circle on blue 😁

File: static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js (hunk 6 above)
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js]: but could still do the same to line 55, could be just `...'click', complete)`

File: static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js (hunk 4 above)
Comment (rtyley) [inline static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js]: https://github.com/guardian/frontend/blob/3194e47/static/src/javascripts/projects/common/modules/experiments/ab.js#L111-L113 seems to indicate that `showForSensitive` is keyed off `shouldHideAdverts`, which sounds reasonable - I'll take out the `pageObj.isSensitive` clause.

File: static/src/javascripts/projects/common/modules/experiments/ab.js (hunk 3 above)
Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/experiments/ab.js]: Yes, but please do rename to contributions in a separate PR 😄

File: docs/ab-testing.md [hunk 8]
```diff
@@ -165,7 +165,7 @@ You can stop and start the test using our [switchboard](https://frontend.gutools
 
 To see the test in action locally: `localhost:9000/[articleId]#ab-[testName]=[variantName]`
 
-To see the tests you are part of using Google Chrome: open the Dev Console -> Resources -> Local Storage -> choose `http://[domain]` (e.g. localhost or www.theguardian.com)  -> check the `gu.ab.participations` row.
+To see the tests you are part of using Google Chrome: open the Dev Console -> Resources -> Local Storage -> choose `http://[domain]` (e.g. localhost or www.theguardian.com)  -> check the `gu.ab.participations` row. Alternatively, if you're using Chrome, use the [extension](https://chrome.google.com/webstore/detail/guardian-ab-tests/nehbenedinjacnhlkjbdneedibcagmno).
```
Comment (johnduffell) [inline docs/ab-testing.md]: good one!

File: static/src/inline-svgs/contributions/coin.svg [hunk 9]
```diff
@@ -0,0 +1 @@
+<svg width="18" height="18" viewBox="5 5 18 18"><defs><circle id="a" cx="14" cy="14" r="9"/></defs><use overflow="visible" fill="#FB0"/><clipPath id="b"><use overflow="visible"/></clipPath><circle clip-path="url(#b)" fill="none" stroke="#FF9B0B" stroke-miterlimit="10" stroke-dasharray="2,2" cx="14" cy="14" r="7.615"/><g clip-path="url(#b)"><defs><path id="c" d="M11.64 14.604h.655v.593c0 .86-.296 1.237-.818 1.515v.89h4.902l.142-1.217h-3.478v-.03c.45-.33.726-.676.777-1.484l.01-.265h1.75v-.942h-1.687l.05-.665c.072-1.104.623-1.35 1.33-1.35.46 0 .798.07 1.145.194v-.962c-.39-.296-.89-.48-1.75-.48-1.36 0-2.373.777-2.373 2.404v.86h-.654v.94z"/></defs><clipPath id="d"><use overflow="visible"/></clipPath><g clip-path="url(#d)"><defs><path id="e" d="M-55.676-2.755h986.54v542.77h-986.54z"/></defs><clipPath id="f"><use overflow="visible"/></clipPath><path clip-path="url(#f)" fill="#C05303" d="M8.016 6.937h11.967v14.126H8.016z"/></g></g></svg>
```
Comment (johnduffell) [inline static/src/inline-svgs/contributions/coin.svg]: 💰

**Comment by johnduffell** [comment]
looks great, will be an interesting test, although I'm with zef on the logo, it looks a bit out of place for some reason.  would be interesting to do further testing on the exact wording too
👍 for the idea

File: common/app/conf/switches/ABTestSwitches.scala [hunk 10]
```diff
@@ -63,4 +63,14 @@ trait ABTestSwitches {
     sellByDate = new LocalDate(2016, 8, 31),
     exposeClientSide = true
   )
+
+  val ABGiraffeHeader20160802 = Switch(
```
Comment (NataliaLKB) [inline common/app/conf/switches/ABTestSwitches.scala]: Could we rename the test too?

**Comment by rtyley** [comment]
I think I've addressed all the feedback in this PR (apart from the logo design itself - Alex B did sign-off on this, but obviously it's going to be worth iterating on) - can I proceed with merging?

**Comment by NataliaLKB** [comment]
I have one more comment, and also agree with @zeftilldeath 's recommendation regarding the design. After that 👍
